### PR TITLE
Enable foreign key constraints by default for Sqlite backend

### PR DESCRIPTION
**Pull request by rdnetto**
This is the PR from #646 re-raised.

CC: @snoyberg

### CODE REVIEW

**Comment by snoyberg** [comment]
This constitutes significantly breaking changes to the API. Personally, I'm very averse to such changes unless they are absolutely warranted. I thought we'd decided on keeping the existing API and extending with new functions. Am I misremembering?

**Comment by rdnetto** [comment]
I think some of the details might have gotten lost in the discussion.

Reading over it again, we agreed on the `mkSqliteSettings` approach (which I ended up calling `mkSqliteConnectionInfo`), but there was also a comment about having new functions to avoid breaking API which I missed.

I'll add a commit to fix this up so that it doesn't break API.

**Comment by snoyberg** [comment]
Awesome, thanks!

**Comment by rdnetto** [comment]
OK, that should do it. We're now exporting new functions, with the old ones marked as deprecated.

File: persistent-sqlite/ChangeLog.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.6.1.1
```
Comment (snoyberg) [inline persistent-sqlite/ChangeLog.md]: This is going to need to be 2.6.2, but I can change that during merging.

File: persistent-sqlite/Database/Persist/Sqlite.hs [hunk 2]
```diff
@@ -57,7 +68,17 @@ import Control.Monad.Trans.Writer (runWriterT)
 -- Instead, use 'withSqliteConn'.
 createSqlitePool :: (MonadIO m, MonadLogger m, MonadBaseControl IO m, IsSqlBackend backend)
                  => Text -> Int -> m (Pool backend)
-createSqlitePool s = createSqlPool $ open' s
+createSqlitePool = createSqlitePoolFromInfo . conStringToInfo
+{-# DEPRECATED createSqlitePool "Use createSqlitePoolFromInfo" #-}
```
Comment (snoyberg) [inline persistent-sqlite/Database/Persist/Sqlite.hs]: I'm not sure what the motivation is for deprecating these, I don't think we discussed that either. These string-based functions are likely sufficient for most use cases.

File: persistent-sqlite/Database/Persist/Sqlite.hs [hunk 3]
```diff
@@ -429,9 +482,9 @@ escape (DBName s) =
     go '"' = "\"\""
     go c = T.singleton c
 
--- | Information required to connect to a sqlite database
+-- | Information required to setup a connection pool.
 data SqliteConf = SqliteConf
-    { sqlDatabase :: Text
+    { sqlConnInfo :: SqliteConnectionInfo
```
Comment (snoyberg) [inline persistent-sqlite/Database/Persist/Sqlite.hs]: This is a breaking change.

* [ ] Resolve breaking change

File: persistent-sqlite/Database/Persist/Sqlite.hs (hunk 2 above)
Comment (rdnetto) [inline persistent-sqlite/Database/Persist/Sqlite.hs]: - [ ] Remove deprecations

File: persistent-sqlite/Database/Persist/Sqlite.hs [hunk 4]
```diff
@@ -484,19 +481,29 @@ escape (DBName s) =
...
     , sqlPoolSize :: Int
     } deriving Show
 
 instance FromJSON SqliteConf where
-    parseJSON v = modifyFailure ("Persistent: error loading Sqlite conf: " ++) $
-      flip (withObject "SqliteConf") v $ \o -> SqliteConf
-        A.<$> o .: "database"
-        A.<*> o .: "poolsize"
+    parseJSON v = modifyFailure ("Persistent: error loading Sqlite conf: " ++) $ flip (withObject "SqliteConf") v parser where
+        parser o = if HashMap.member "database" o
```
Comment (snoyberg) [inline persistent-sqlite/Database/Persist/Sqlite.hs]: This works fine, but for future reference the `Alternative` instance would probably be easier:

```haskell
(SqliteConf <$>o .: "database" ...) <|> (SqliteConfInfo <$> o.: "connInfo" ...)
```

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**